### PR TITLE
refactor: update @azure/ms-rest-nodeauth

### PR DESCRIPTION
**Pull request by IanChokS**
Fix for #1146 

Refactors `.loginWithAppServiceMSI()` function config to include `clientId` for user-assigned MSI Auth

### CODE REVIEW

File: src/connection.ts [hunk 1]
```diff
@@ -3549,9 +3548,8 @@ Connection.prototype.STATE = {
               }, getTokenFromCredentials);
             } else if (authentication.type === 'azure-active-directory-msi-app-service') {
               loginWithAppServiceMSI({
-                msiEndpoint: authentication.options.msiEndpoint,
-                msiSecret: authentication.options.msiSecret,
-                resource: fedAuthInfoToken.spn
+                resource: fedAuthInfoToken.spn,
+                clientId: authentication.options.clientId
```
Comment (arthurschreiber) [inline src/connection.ts:3552]: Is there a specific reason for dropping the `msiEndpoint` and `msiSecret` options here? 🤔

Comment (IanChokS) [reply src/connection.ts:3552]: It doesn't appear to be needed according to the [docs](https://github.com/Azure/ms-rest-nodeauth) 
"`The code below works for both system managed and user-assigned managed identities. You can leave the options empty if you want to use system managed identity. If you want to use the user-assigned managed identity, you must at least provide the clientId in the options.`"

Comment (arthurschreiber) [reply src/connection.ts:3552]: That is correct, and that is what made user identities not work correctly here. But the `loginWithAppServiceMSI` still has support for the `msiEndpoint` and `msiSecret`, and I think we want to keep supporting those two options, in addition to `clientId`. 🙇‍♂️

Comment (IanChokS) [reply src/connection.ts:3552]: Ok 👍

**Review by sharonchoong — APPROVED** [review]
Tested that the user-assigned MSI authentication worked with this PR.  Haven't tested all Azure authentication types but password authentication also had no issues.

**Review by MichaelSun90 — APPROVED** [review]

**Comment by 72636c** [comment]
Hiya @IanChokS, should this be a `fix` given it changes `dependencies`?

A new release containing this PR would be great as it drops a transitive `axios` dependency that is flagged by vulnerability scanners: https://snyk.io/vuln/SNYK-JS-AXIOS-1038255

**Comment by arthurschreiber** [comment]
This upgrade is causing a `UnhandledPromiseRejectionWarning: ConnectionError: Security token could not be authenticated or authorized.` error when trying to connect via Azure AD password authentication. I'm not sure what's going on exactly, but I'll try to figure out what's going on. 🤔

**Pull request merged**